File: confirmEnding.ts
const confirmEnding = (source: string, endWord: string) => {
  const isString = (value: unknown): value is string => typeof value === "string" && value.trim().length > 0;
  if (!isString(source) || !isString(endWord)) {
    throw new Error("Both source and endWord must be valid strings");
  }
  return source.endsWith(endWord);
}

console.log(confirmEnding("Ricardo Manuel Santana Saumeth", "Saumeth"))
console.log(confirmEnding("Ricardo Manuel Santana Saumeth", "SaumethX"))
console.log(confirmEnding("", ""))

File: confirmEnding.py
def confirm_ending(target: str, suffix: str) -> bool:
    # Input validation
    if not isinstance(target, str) or not isinstance(suffix, str):
        raise ValueError("Both inputs must be strings")

    # Handle empty strings
    if not suffix and not target:
        return True

    return target.endswith(suffix)


print(confirm_ending("Ricardo Manuel", "Manuelx"))
